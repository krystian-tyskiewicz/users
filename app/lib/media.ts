const media = {
  phone: '@media (max-width: 480px)',
  tablet: '@media (max-width: 768px)',
}

export default media
