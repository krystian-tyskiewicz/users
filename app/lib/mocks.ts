export const user = {
  id: '1234',
  name: 'John Doe',
  username: 'john-doe',
  email: 'john@doe.com',
  city: 'New York',
}
