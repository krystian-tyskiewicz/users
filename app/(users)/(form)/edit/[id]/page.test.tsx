import { render } from '@testing-library/react'

import Page from './page'
import { user } from '@/app/lib/mocks'

jest.mock('../../../../redux/services/users', () => ({
  useGetUserQuery: jest.fn().mockImplementation(() => ({
    isLoading: false,
    data: user,
  })),
  useEditUserMutation: jest.fn().mockReturnValue([])
}))
jest.mock('next/navigation', () => ({
  useRouter: jest.fn(),
}))

it('renders correctly', () => {
  const { container } = render(<Page params={{ id: '1234' }} />)
  expect(container).toMatchSnapshot()
})
