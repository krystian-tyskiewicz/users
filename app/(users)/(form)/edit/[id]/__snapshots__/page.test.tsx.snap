// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <header
    class="sc-jRQBzY bPfapj"
  >
    <h4
      class="sc-bqiRFl eSiAwy"
    >
      Edit user
    </h4>
  </header>
  <form
    class="sc-ksdynI"
  >
    <div
      class="sc-eCIlHX kAWkuI"
    >
      <input
        class="sc-gKclGN kytkRg"
        name="name"
        placeholder="Name*"
        value="John Doe"
      />
    </div>
    <div
      class="sc-eCIlHX kAWkuI"
    >
      <input
        class="sc-gKclGN kytkRg"
        name="username"
        placeholder="Username*"
        value="john-doe"
      />
    </div>
    <div
      class="sc-eCIlHX kAWkuI"
    >
      <input
        class="sc-gKclGN kytkRg"
        name="email"
        placeholder="Email*"
        value="john@doe.com"
      />
    </div>
    <div
      class="sc-eCIlHX kAWkuI"
    >
      <input
        class="sc-gKclGN kytkRg"
        name="city"
        placeholder="City"
        value="New York"
      />
    </div>
    <div
      class="sc-eCIlHX kAWkuI"
    >
      <div
        class="sc-gsDKeI ejuFsk"
      >
        <button
          class="sc-bdvumH ynMys sc-dkPulx lgyEdo"
          type="button"
        >
          Cancel
        </button>
        <button
          class="sc-bdvumH ftCnfd sc-dkPulx lgyEdo"
          type="submit"
        >
          Submit
        </button>
      </div>
    </div>
  </form>
</div>
`;
